src/pin.ts

```diff
@@ -20,23 +20,34 @@ interface CheloniaConfig {
   contracts: {
     [contractName: string]: {
       version: string
-      path: string // relative path to the contract file
+      path: string // relative path to the manifest file
     }
   }
   // Future: can add other Chelonia-specific configuration here
   // like deployment targets, signing keys, etc.
 }
 
+/**
+ * Contract file information extracted from manifest
+ */
+interface ContractFiles {
+  main: string
+  slim?: string
+}
+
 /**
  * ContractPinner class handles pinning individual contracts to specific versions
- * 1. Uses chelonia.json for Chelonia-specific configuration (separate from Node.js package.json)
- * 2. Supports independent versioning for each contract (contracts can have different versions)
- * 3. Organizes contracts in a structured directory layout: contracts/<name>/<version>/<files...>
- * 4. Performs selective updates - only modifies the specific contract being pinned
- * 5. Maintains compatibility with chel manifest and chel deploy workflows
  *
- * The pinning process copies contract files from a baseline version to create new versions,
- * allowing developers to iterate on contracts while maintaining version history.
+ * SECURITY FIX: This implementation takes the manifest file directly as input
+ * instead of extracting contract names from user-controlled file paths and
+ * using unsafe regular expressions. This addresses @corrideat's security feedback.
+ *
+ * Key improvements:
+ * 1. Takes manifest file path directly as input
+ * 2. Extracts contract name and file paths from the manifest content
+ * 3. Uses chelonia.json for Chelonia-specific configuration
+ * 4. Organizes contracts in structured directory layout: contracts/<name>/<version>/
+ * 5. Maintains compatibility with chel manifest and chel deploy workflows
```

Comment:
The previous comment was mostly fine. The previous one is appropriate maybe for a changelog but not in the source file, as it discusses changes without context.

Reply:
Should I revert back to previous one?